packages/simple-markdown/src/index.js

```diff
@@ -410,6 +410,12 @@ var parserFor = function (rules: ParserRules, defaultState: ?State): Parser {
                 // $FlowFixMe
                 Array.prototype.push.apply(result, parsed);
             } else {
+                if (parsed == null || typeof parsed !== "object") {
+                    throw new Error(
+                        `parse() function returned invalid parse result: '${parsed}'`,
+                    );
+                }
```

Comment:
This isn't being covered in the tests.  Having a test for this would be nice.

Reply:
Good idea. Added!